server/routes/api/notifications/notifications.ts

```diff
@@ -49,4 +57,121 @@ router.post(
   handleUnsubscribe
 );
 
+router.post(
+  "notifications.list",
+  auth(),
+  pagination(),
+  validate(T.NotificationsListSchema),
+  transaction(),
+  async (ctx: APIContext<T.NotificationsListReq>) => {
+    const eventType = ctx.input.body.eventType ?? ctx.input.query.eventType;
+    const archived = ctx.input.body.archived ?? ctx.input.query.archived;
+    const user = ctx.state.auth.user;
+    let where: WhereOptions<Notification> = {
+      userId: user.id,
+    };
+    if (eventType) {
+      where = { ...where, event: eventType };
+    }
+    if (archived) {
+      where = {
+        ...where,
+        archivedAt: {
+          [Op.ne]: null,
+        },
+      };
+    }
+    const [notifications, total] = await Promise.all([
+      Notification.scope(["withUser", "withActor"]).findAll({
+        where,
+        order: [["createdAt", "DESC"]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+      }),
+      Notification.count({
+        where: {
+          ...where,
+          viewedAt: {
+            [Op.is]: null,
+          },
+        },
```

Comment:
```suggestion
        where,
```

Unfortunately I suppose we need two count operations here, the count for total which should reflect the total for the filtered list for pagination and the unviewed total separately?

Reply:
Sure.